Taskfile.yml

```diff
@@ -156,23 +157,29 @@ tasks:
   asoctl:lint:
     desc: Run {{.ASOCTL_APP}} fast lint checks.
     dir: '{{.ASOCTL_ROOT}}'
-    deps: [controller:generate-crds]
     cmds:
       - golangci-lint run --verbose --fast=false --timeout 5m ./...
 
   asoctl:build:
     desc: Generate the {{.ASOCTL_APP}} binary.
     dir: '{{.ASOCTL_ROOT}}'
     label: asoctl:build-{{.GOOS}}-{{.GOARCH}}
-    deps: [controller:generate-crds]
     generates:
-      - ../../bin/{{.ASOCTL_APP}}-{{.GOOS}}-{{.GOARCH}}
-    cmds:
-      - GOOS={{.GOOS}} GOARCH={{.GOARCH}} go build {{.VERSION_FLAGS}} -o ../../bin/{{.ASOCTL_APP}}-{{.GOOS}}-{{.GOARCH}}{{.EXT}} .
+      - "{{.ARCHIVE}}"
+    cmds:
+      - mkdir -p ./bin/{{.GOOS}}-{{.GOARCH}}
+      - GOOS={{.GOOS}} GOARCH={{.GOARCH}} go build {{.VERSION_FLAGS}} -o {{.EXECUTABLE}}
+      - if [ "{{.ARCHIVETYPE}}" = ".zip" ]; then zip -j -r {{.ARCHIVE}} {{.EXECUTABLE}}; fi
+      - if [ "{{.ARCHIVETYPE}}" = ".gz" ]; then gzip -v -c {{.EXECUTABLE}} > {{.ARCHIVE}} ; fi
+      # gzip -c ./bin/darwin-arm64/asoctl > ./asoctl.gz
+      #../../bin/{{.ASOCTL_APP}}-{{.GOOS}}-{{.GOARCH}}{{.EXT}} .
     vars:
       EXT: '{{if eq .GOOS "windows"}}.exe{{else}}{{end}}'
+      ARCHIVETYPE: '{{if eq .GOOS "windows"}}.zip{{else}}.gz{{end}}'
       GOOS: '{{default OS .GOOS}}'
       GOARCH: '{{default ARCH .GOARCH}}'
+      EXECUTABLE: ./bin/{{.GOOS}}-{{.GOARCH}}/{{.ASOCTL_APP}}{{.EXT}}
+      ARCHIVE: ../../bin/{{.ASOCTL_APP}}-{{.GOOS}}-{{.GOARCH}}{{.ARCHIVETYPE}}
```

Comment:
Do we have documentation about how to install asoctl? I think we do. Can we fix up that documentation as part of this PR? Or if we're worried about that documentation then being wrong for a while until we release 2.2.0 can we file a high-priority issue in 2.2.0 to ensure we do that before release (but closer to release)?

Reply:
Logged a documentation issue - see #3108